### PR TITLE
docs(django) - avoid auto_now_add

### PR DESCRIPTION
**Pull request by hazelw**

### CODE REVIEW

File: languages/Python/architecture.md [hunk 1]
```diff
@@ -156,4 +156,26 @@ class Entity(models.Model):
...
+
+Do:
+```python
+class Snake(models.Model):
+    sighted_at = models.DateTimeField(default=timezone.now)
+```
+
+```python
+class Snake(models.Model):
+    sighted_at = models.DateTimeField(default=tz_now)
```
Comment (martyphee) [inline languages/Python/architecture.md:180]: I'm not sure about these last examples. `tz_now` is `Europe/London` timezone form what I've seen. Are the servers set to London also so that `timezone.now` would be `Europe/London`?

Generally, when setting dates in a table you always use UTC.

File: languages/Python/architecture.md [hunk 2]
```diff
@@ -156,4 +156,26 @@ class Entity(models.Model):
 ```
 
 
+### Avoid auto_now_add=True
+Django provides an `auto_now_add` option for `DateField` and `DateTimeField`. By setting this field, Django implicitly marks the field as not editable and any attempts to write to it will silently fail.
```
Comment (martyphee) [inline languages/Python/architecture.md:160]: I like this behaviour. If you added a `created_at` field to your model I definitely do not want anyone to be able to set the value from what it was originally set to.

File: languages/Python/architecture.md [hunk 3]
```diff
@@ -156,4 +156,26 @@ class Entity(models.Model):
...
 
+##### Why:
+- As the field is marked as not editable, this means that it isn't editable in your tests, either. `auto_now_add=True` means that Django will silently ignore any attempts to make a change to that field, regardless of whether the entity attempting to make the change is your code, your test's setup phase, or even a factory attempting to set a default datetime.
```
Comment (martyphee) [inline languages/Python/architecture.md:163]: Should you be able to edit this field in a test? If I'm counting on the framework to set this value then I shouldn't be editing it. After creation, I would validate that the field has been set. If you need a specific timeframe wouldn't you use `freezegun`?

File: languages/Python/architecture.md (hunk 1 above)
Comment (gregbeech) [reply languages/Python/architecture.md:180]: 💯 timestamps in databases should just represent which point in time they refer to; time zones are for display purposes only!

File: languages/Python/architecture.md (hunk 2 above)
Comment (gregbeech) [reply languages/Python/architecture.md:160]: If it was "any attempts to write to it will raise an error" then I'd probably agree with you, but if it silently fails without setting the value then that feels like a recipe for endless days lost debugging annoying issues.

Comment (martyphee) [reply languages/Python/architecture.md:160]: Let me rephrase that then. Only use `auto_now_add` on `created_at` fields.

Comment (gregbeech) [reply languages/Python/architecture.md:160]: Even so, I think you'd still find the problem of just trying to set it in tests rather than using Freezegun and it just not working and you tearing your hair out until you found out why!

Comment (hazelw) [reply languages/Python/architecture.md:160]: to confirm have spent at least 4 hours of my life tearing out my hair over this :D

File: languages/Python/architecture.md (hunk 1 above)
Comment (hazelw) [reply languages/Python/architecture.md:180]: I want this to be true, but it's not what we're currently doing. @stuartleigh any idea why we're storing non-UTC datetimes?
